### DISCUSSION TITLE
New dark and light attribute

### DISCUSSION THREAD

**Discussion by New-King** [tailwindlabs/tailwindcss#12266] [General]
```
<div class="bg-white dark:bg-slate-800 rounded-lg px-6 py-8 ring-1 ring-slate-900/5 shadow-xl text-black dark:text-white">
```

```
<div 
class="rounded-lg px-6 py-8 ring-1 ring-slate-900/5 shadow-xl"
light="bg-white text-black"
dark="bg-slate-800 text-white" 
>
```

Is there a possibility or a plan? I really need this pattern

**Comment by alex-krasikau** [comment]
Hey @New-King!

This doesn't appear to be a good pattern. What Tailwind CSS does is generate a CSS file with CSS rules. CSS rules are applied to the HTML elements that match the CSS selector. Most developers use classes to define CSS selectors, and it's really beneficial to stick to this practice.

Even if you manage to generate the `[dark="bg-slate-800"]` selector in the HTML, it will produce a much larger CSS output file, and it won't align with common development patterns. This can be problematic when you start collaborating with other developers on this or any future project.

To make it easier to work with the dark:bg-slate-800 modifier, I would recommend using the [Tailwind Prettier Plugin](https://tailwindcss.com/blog/automatic-class-sorting-with-prettier), which sorts classes in a user-friendly way.

**Reply by New-King** [reply]
Thank you for your reply！
